dds_web/web/user.py

```diff
@@ -148,6 +152,81 @@ def register():
     return flask.render_template("user/register.html", form=form)
 
 
+@auth_blueprint.route("/activate_totp/<token>", methods=["GET", "POST"])
+@limiter.limit(
+    dds_web.utils.rate_limit_from_config,
+    methods=["GET", "POST"],
+    error_message=ddserr.TooManyRequestsError.description,
+)
+@flask_login.login_required
+def activate_totp(token):
+    user = flask_login.current_user
+
+    form = forms.ActivateTOTPForm()
+
+    dds_web.security.auth.verify_activate_totp_token(token, current_user=user)
+
+    if user.totp_enabled:
+        flask.flash("Two-factor authentication via authenticator app is already enabled.")
+        return flask.redirect(flask.url_for("pages.home"))
+
+    # Don't change secret on page reload
+    if not user.totp_initiated:
```

Comment:
Should there be an else for this too? Or is the idea that it's fine if it's already initiated?

Reply:
Also, when is totp_initiated set to True?